andela/denethor-ah-backend
# #163478812 Users should be able to bookmark articles for reading later, delete bookmark and get all bookmarks by self

#### What does this PR do?
> Create, Get and Delete Bookmarks of an Article
> 
 #### Description of Task to be completed?
> • Create Bookmark routes to `create` and `delete` bookmarks
> • Create Bookmark routes to `get` all bookmarks of a user
> • Create controller methods to perform the above actions
> • Modify related model/migration files and associations
> • Write tests
> 
 #### How should this be manually tested?
> • Clone this branch
> • CD into the repo
> • Run `npm install`
> • Run `npm run migrate && npm run seed`
> • Run `npm run dev`
> • Use postman to check the following endpoints:
> 
> ```
>  • POST `http://localhost/5000/articles/:articleId/bookmark`;
>  • DELETE `http://localhost/5000/articles/:articleId/bookmark `;
>  • GET ` http://localhost/5000/users/:userId/bookmarks`;

> ```
 #### Background contexts?
> Note that you must be an authenticated user to perform endpoint operations.

> ```
> • Use generated token as `bearer token` auth type to access endpoints.
> 
#### What are the relevant pivotal tracker stories?
 [Users should be able to bookmark articles for reading later, delete bookmark and get all bookmarks by self](https://www.pivotaltracker.com/story/show/163478812)
